zeebe/gateway-rest/src/main/java/io/camunda/zeebe/gateway/rest/SearchQueryRequestMapper.java

```diff
@@ -159,50 +159,51 @@ private static Either<List<String>, DecisionDefinitionFilter> toDecisionDefiniti
     return Either.right(builder.build());
   }
 
-  public static Either<ProblemDetail, List<VariableValueFilter>> toVariableValueFilter(
+  public static List<VariableValueFilter> toVariableValueFilter(
       final List<VariableValueFilterRequest> filters) {
 
     final List<VariableValueFilter> result;
 
     if (filters != null && !filters.isEmpty()) {
-      result =
-          filters.stream()
-              .map(SearchQueryRequestMapper::toVariableValueFilter)
-              .map(Either::get)
-              .toList();
+      result = filters.stream().map(SearchQueryRequestMapper::toVariableValueFilter).toList();
     } else {
       result = null;
     }
-
-    return Either.right(result);
+    return result;
   }
 
-  private static Either<ProblemDetail, VariableValueFilter> toVariableValueFilter(
-      final VariableValueFilterRequest f) {
-    return Either.right(
-        FilterBuilders.variableValue(
-            (v) ->
-                v.name(f.getName())
-                    .eq(f.getEq())
-                    .gt(f.getGt())
-                    .gte(f.getGte())
-                    .lt(f.getLt())
-                    .lte(f.getLte())));
+  private static VariableValueFilter toVariableValueFilter(final VariableValueFilterRequest f) {
+    return FilterBuilders.variableValue(
+        (v) ->
+            v.name(f.getName())
+                .eq(f.getEq())
+                .gt(f.getGt())
+                .gte(f.getGte())
+                .lt(f.getLt())
+                .lte(f.getLte()));
   }
 
   public static Either<List<String>, SearchQueryPage> toSearchQueryPage(
       final SearchQueryPageRequest requestedPage) {
-    if (requestedPage != null) {
-      return Either.right(
-          SearchQueryPage.of(
-              (p) ->
-                  p.size(requestedPage.getLimit())
-                      .from(requestedPage.getFrom())
-                      .searchAfter(toArrayOrNull(requestedPage.getSearchAfter()))
-                      .searchBefore(toArrayOrNull(requestedPage.getSearchBefore()))));
+    if (requestedPage == null) {
+      return Either.right(null);
     }
 
-    return Either.right(null);
+    final Object[] searchAfter = toArrayOrNull(requestedPage.getSearchAfter());
+    final Object[] searchBefore = toArrayOrNull(requestedPage.getSearchBefore());
+
+    if (searchAfter != null && searchBefore != null) {
+      return Either.left(
+          List.of("Error: Both searchAfter and searchBefore cannot be set at the same time."));
```

Comment:
👍 I like this new validation 🚀 
🔧 Let's please add all error messages to the `ErrorMessages` class so they are centralized. Please also remove the `.` at the end of the message, those will be added by default to all error messages 🙂